### PR TITLE
Updated the compatibility of version 1.3. * And 1.4. * Of SQLAlchemy 

### PR DESCRIPTION
**Pull request by jssuzanne**
see #16

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-sqlalchemy==1.3.17
+sqlalchemy
```
Comment (jklukas) [inline requirements.txt]:
```suggestion
sqlalchemy==1.4.4
```

I'd prefer all `requirements.txt` entries stay pinned for stability of testing.

File: sqlalchemy_views/views.py [hunk 2]
```diff
@@ -32,7 +32,13 @@ class CreateView(_CreateDropBase):
 
     def __init__(self, element, selectable, on=None, bind=None,
                  or_replace=False, options=None):
-        super(CreateView, self).__init__(element, on=on, bind=bind)
+        try:
+            super(CreateView, self).__init__(element, on=on, bind=bind)
+        except TypeError:
+            # Since version 1.4.0 of SQLAlchemy the ** on ** parameter no
+            # longer exists. it causes a ** TypeError ** exception
+            super(CreateView, self).__init__(element, bind=bind)
```
Comment (jklukas) [inline sqlalchemy_views/views.py:43]: We should probably add a test here for `if on is not None` and raise an exception specifying that `on` is not supported on SQLAlchemy 1.4+. I'm concerned about silently ignoring `on` once folks upgrade to SQA 1.4.

File: sqlalchemy_views/views.py [hunk 3]
```diff
@@ -83,7 +92,13 @@ class DropView(_CreateDropBase):
 
     def __init__(self, element, on=None, bind=None,
                  cascade=False, if_exists=False):
-        super(DropView, self).__init__(element, on=on, bind=bind)
+        try:
+            super(DropView, self).__init__(element, on=on, bind=bind)
+        except TypeError:
+            # Since version 1.4.0 of SQLAlchemy the ** on ** parameter no
+            # longer exists. it causes a ** TypeError ** exception
+            super(DropView, self).__init__(element, bind=bind)
```
Comment (jklukas) [inline sqlalchemy_views/views.py:106]: We should probably add a test here for `if on is not None` and raise an exception specifying that on is not supported on SQLAlchemy 1.4+. I'm concerned about silently ignoring on once folks upgrade to SQA 1.4.

**Review by jklukas — CHANGES_REQUESTED** [review]
Thanks for the patch! This looks great. I just have the one concern about checking for `on` being set. If you can address that, I'll get this merged and I'll make a new release.

**Review by jklukas — APPROVED** [review]

**Comment by jssuzanne** [comment]
Thank you for the review and for the merge

**Comment by jklukas** [comment]
> Thank you for the review and for the merge

Thanks for the contribution! I'll plan to make a release later today.

**Comment by rad-pat** [comment]
Before you release, we perhaps have a better resolution for the missing `on` kwarg - just pass through the kwargs specified, then it will also be backwards compatible.  https://github.com/PlaidCloud/sqlalchemy-views/commit/6e7147f00233e446dfb5206a30428ffd601fe89b

**Comment by jklukas** [comment]
I just made the 0.3.0 release, but I wouldn't be opposed to releasing again if we have a better solution. 

I can see the benefit of the https://github.com/PlaidCloud/sqlalchemy-views/commit/6e7147f00233e446dfb5206a30428ffd601fe89b but it could potentially break existing use cases. It's possible for users to have specified `on` as a positional param rather than via keyword. So I would be open to adding `**kw` here as a solution for future param changes, but I don't think we should remove `on` from the existing parameter list.

**Comment by jssuzanne** [comment]
We can also add a deprecated message if on is None
